src/software/ai/hl/stp/play/defense_play_test.py

```diff
@@ -0,0 +1,221 @@
+
+    simulated_test_runner.blue_full_system_proto_unix_io.send_proto(Play, blue_play)
+    simulated_test_runner.yellow_full_system_proto_unix_io.send_proto(Play, yellow_play)
+
+    # Create world state
+    simulated_test_runner.simulator_proto_unix_io.send_proto(
+        WorldState,
+        create_world_state(
+            yellow_robot_locations=yellow_bots,
+            blue_robot_locations=blue_bots,
+            ball_location=ball_initial_pos,
+            ball_velocity=tbots.Vector(0, 0),
+        ),
+    )
+
+    # Always Validation
+    # TODO- #2753 Validation
+    always_validation_sequence_set = [[]]
+
+    # Eventually Validation
+    # TODO- #2753 Validation
+    eventually_validation_sequence_set = [[]]
+
+    simulated_test_runner.run_test(
+        eventually_validation_sequence_set=eventually_validation_sequence_set,
+        always_validation_sequence_set=always_validation_sequence_set,
+        test_timeout_s=10,
+    )
+
+
+def test_defense_play_close_to_net(simulated_test_runner):
+
+    # starting point must be Point
+    ball_initial_pos = tbots.Point(0.9, 2.85)
+
+    # Setup Bots
+    blue_bots = [
+        tbots.Point(-4.5, 0),
+        tbots.Point(-3, 1.5),
+        tbots.Point(-3, 0.5),
+        tbots.Point(-3, -0.5),
+        tbots.Point(-3, -1.5),
+        tbots.Point(-3, -3.0),
+    ]
+
+    yellow_bots = [
+        tbots.Point(-2, 1.05),
+        tbots.Point(-3.5, 2),
+        tbots.Point(-1.5, 0),
+        tbots.Point(-2.3, -1),
+        tbots.Point(-3.8, -2),
+        tbots.Field.createSSLDivisionBField().enemyGoalCenter(),
+    ]
+
+    # Game Controller Setup
+    simulated_test_runner.gamecontroller.send_ci_input(
+        gc_command=Command.Type.STOP, team=Team.UNKNOWN
+    )
+    simulated_test_runner.gamecontroller.send_ci_input(
+        gc_command=Command.Type.FORCE_START, team=Team.BLUE
+    )
+
+    # Force play override here
+    blue_play = Play()
+    blue_play.name = PlayName.DefensePlay
+
+    yellow_play = Play()
+    yellow_play.name = PlayName.HaltPlay
+
+    simulated_test_runner.blue_full_system_proto_unix_io.send_proto(Play, blue_play)
+    simulated_test_runner.yellow_full_system_proto_unix_io.send_proto(Play, yellow_play)
+
+    # Create world state
+    simulated_test_runner.simulator_proto_unix_io.send_proto(
+        WorldState,
+        create_world_state(
+            yellow_robot_locations=yellow_bots,
+            blue_robot_locations=blue_bots,
+            ball_location=ball_initial_pos,
+            ball_velocity=tbots.Vector(0, 0),
+        ),
+    )
+
+    # Always Validation
+    # TODO- #2753 Validation
+    always_validation_sequence_set = [[]]
+
+    # Eventually Validation
+    # TODO- #2753 Validation
+    eventually_validation_sequence_set = [[]]
+    simulated_test_runner.run_test(
+        eventually_validation_sequence_set=eventually_validation_sequence_set,
+        always_validation_sequence_set=always_validation_sequence_set,
+        test_timeout_s=10,
+    )
```

Comment:
looks like the varying aspects of these tests are the blue and yellow robot starting positions. let's parametrize them

Reply:
bump